tests/trainer/logging_/test_eval_loop_logging.py

```diff
@@ -590,3 +590,29 @@ def get_metrics_at_idx(idx):
         "test_loss",
     }
     assert set(results[0]) == {"test_loss"}
+
+
+def test_logging_dict_on_validation_step(tmpdir):
+    class TestModel(BoringModel):
+        def validation_step(self, batch, batch_idx):
+            loss = super().validation_step(batch, batch_idx)
+            loss = loss["x"]
+            metrics = {
+                "loss": loss,
+                "loss_1": loss,
+            }
+            self.log("val_metrics", metrics)
+
+        validation_epoch_end = None
+
+    model = TestModel()
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_train_batches=2,
+        limit_val_batches=2,
+        max_epochs=2,
```

Comment:
```suggestion
        num_sanity_val_batches=2,
        max_epochs=1,
```
To be explicit about what causes the bug

Reply:
This line https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/connectors/logger_connector/result.py#L595 + default metadata=None in ResultMetricCollection.

As apply_collection was re-creating the object and metadata wasn't passed to the init function, it just override itself with metadata and broke on next self.log.

I believe it is safer to removed entirely and depend on its children.